dataprofiler/profilers/profile_builder.py

```diff
@@ -1442,25 +1442,40 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
-    def _get_correlation(self, clean_samples):
+    def _get_correlation(self, clean_samples, batch_properties):
         """
         Calculate correlation matrix on the cleaned data.
 
         :param clean_samples: the input cleaned dataset
         :type clean_samples: dict()
+        :param batch_properties: mean/std/counts of each batch column necessary
+        for correlation computation
+        :type batch_properties: dict()
         """
         columns = self.options.correlation.columns
         clean_column_ids = []
         if columns is None:
             for idx in range(len(self._profile)):
-                if (self._profile[idx].profile['data_type'] not in ['int', 'float']
-                        or self._profile[idx].null_count > 0):
+                data_type = self._profile[idx].\
+                    profiles["data_type_profile"].selected_data_type
+                if data_type not in ["int", "float"]:
                     clean_samples.pop(idx)
                 else:
                     clean_column_ids.append(idx)
 
-        data = pd.DataFrame(clean_samples)
-        data = data.apply(pd.to_numeric, errors='coerce')
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors='coerce')
+        means = {index:mean for index, mean in enumerate(batch_properties['mean'])}
+        data = data.fillna(value=means)
+
+        # Update the counts/std if needed (i.e. if null rows or exist)
+        if (len(data.index) != batch_properties['count']).any():
+            adjusted_stds = np.sqrt(
+                batch_properties['std']**2 * (batch_properties['count'] - 1) \
+                / (len(data.index) - 1)
+            )
+            batch_properties['std'] = adjusted_stds
+        # Set count key to a single number now that everything's been adjusted
+        batch_properties['count'] = len(data.index)
```

Comment:
any reason specifically of the index?

Reply:
I'm not sure what you quite mean, but `data.index` returns all the row indices, so the length of that is the number of total rows in the cleaned data excluding fully NaN rows but including partially NaN rows.

Reply:
as oppose to `len(data)`

Reply:
this works better, forgot you could do this, i'll make those changes